### PR TITLE
Adding nxos_acl module

### PR DESCRIPTION
**Pull request by GGabriele**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

nxos_acl
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/']
```
##### SUMMARY

New module to manages access list entries for ACLs.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @GGabriele. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @GGabriele, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_acl.py [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+version_added: "2.2"
+short_description: Manages access list entries for ACLs.
+description:
+    - Manages access list entries for ACLs.
+extends_documentation_fragment: nxos
+author:
+    - Jason Edelman (@jedelman8)
+    - Gabriele Gerbino (@GGabriele)
+notes:
+    - I(state)=absent removes the ACE if it exists
```
Comment (gundalow) [inline network/nxos/nxos_acl.py]: `C(state=absent)`

File: network/nxos/nxos_acl.py [hunk 2]
```diff
@@ -0,0 +1,810 @@
...
+    - I(state)=delete_acl deleted the ACL if it exists
```
Comment (gundalow) [inline network/nxos/nxos_acl.py]: `I(state=delete_acl)`

File: network/nxos/nxos_acl.py [hunk 3]
```diff
@@ -0,0 +1,810 @@
...
+    - for idempotency, use port numbers for the src/dest port
```
Comment (gundalow) [inline network/nxos/nxos_acl.py]: For...

File: network/nxos/nxos_acl.py [hunk 4]
```diff
@@ -0,0 +1,810 @@
...
+      params like I(src_port1) and names for the well defined protocols
+      for the I(proto) param.
+    - while this module is idempotent in that if the ace as presented in the
```
Comment (gundalow) [inline network/nxos/nxos_acl.py]: Although this module is...

File: network/nxos/nxos_acl.py [hunk 5]
```diff
@@ -0,0 +1,810 @@
...
+      task is identical to the one on the switch, no changes will be made. If
+      there is any difference, what is in Ansible will be pushed (configured
+      options will be overridden).  This is to improve security, but at the
+      same time remember an ACE is removed, then re-added, so if there is a
+      change, the new ACE will be exacty what params you are sending to the
+      module.
+options:
+    seq:
+        description:
+            - sequence number of the entry (ACE)
```
Comment (gundalow) [inline network/nxos/nxos_acl.py]: Caps & full stops please on all `descriptions:`

File: network/nxos/nxos_acl.py [hunk 6]
```diff
@@ -0,0 +1,810 @@
...
+    - I(state)=delete_acl deleted the ACL if it exists
+    - for idempotency, use port numbers for the src/dest port
+      params like I(src_port1) and names for the well defined protocols
+      for the I(proto) param.
+    - while this module is idempotent in that if the ace as presented in the
+      task is identical to the one on the switch, no changes will be made. If
+      there is any difference, what is in Ansible will be pushed (configured
+      options will be overridden).  This is to improve security, but at the
+      same time remember an ACE is removed, then re-added, so if there is a
+      change, the new ACE will be exacty what params you are sending to the
```
Comment (gundalow) [inline network/nxos/nxos_acl.py]: exactly
paramaters

File: network/nxos/nxos_acl.py [hunk 7]
```diff
@@ -0,0 +1,810 @@
...
+      module.
+options:
+    seq:
+        description:
+            - sequence number of the entry (ACE)
+        required: false
+        default: null
+    name:
+        description:
+            - Case sensitive name of the access list (ACL)
```
Comment (gundalow) [inline network/nxos/nxos_acl.py]: Fullstop

**Comment by gundalow** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @GGabriele for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**